packages/ui-components/src/components/text-field/test/text-field.e2e.ts

```diff
@@ -0,0 +1,30 @@
+import { E2EPage, newE2EPage } from '@stencil/core/testing';
+
+describe('Text Field (end-to-end)', () => {
+	let page: E2EPage;
+
+	describe('when renders with default props', () => {
+		beforeEach(async () => {
+			page = await newE2EPage();
+			await page.setContent('<kv-text-field></kv-text-field>');
+		});
+
+		it('should not render a label', async () => {
+			const labelComponent = await page.find('kv-text-field >>> .label');
+			expect(labelComponent).toBeFalsy();
+		});
+	});
+
+	describe('when has a label', () => {
+		beforeEach(async () => {
+			page = await newE2EPage();
+			await page.setContent('<kv-text-field label="Text Field"></kv-text-field>');
+		});
+
+		it('should render label', async () => {
+			const labelComponent = await page.find('kv-text-field >>> .label');
+			expect(labelComponent).toBeTruthy();
+			expect(labelComponent.innerText.toLocaleLowerCase()).toBe('text field');
+		});
+	});
```

Comment:
add more test cases and make sure you add those scenarios to the storybook as well :) 

- when loading check if the div which contains `.input-container-loading` is rendered
- check the help text scenario
- test the blur / textChange events (when enabled and disabled)
- required

if you have any questions shoot them over slack ;)